### PR TITLE
Typing hints

### LINKED ISSUE #6: Stronger typing
I have not yet looked into how to achieve this, but it would be nice if `dicti` was somehow understood by Mypy to be a real subtype of `Dict`. Currently, with strict Mypy settings, it's regarded as `Any`:

```py
def foo() -> Dict[str, str]:
    return dicti({"a": "b"})
```

image

### PR DESCRIPTION
**Pull request by coldfix**
Add typing hints and check using mypy.

Resolves #6

### CODE REVIEW

**Comment by rudolfbyker** [comment]
Thanks for your effort and time! I'm getting this:

image

I'm a bit confused about the correct way forward here, as I have never worked with runtime-generated classes.

**Comment by coldfix** [comment]
Hm, I would guess there are two issues here:

- pydicti declares dicti as a `Mapping` vs your code says it's a `Dict`, which is, I guess, more specific. I should fix that to make `dicti` and `odicti` more specific to `dict` and `OrderedDict`
- your IDE seems to infer the mapping type as `Any -> Any`. This may be a consequence of not being able to infer the type of the `{ ... }` dict argument that you're passing to `dicti`. It might also be a downstream problem due to some other stuff

I'll try to fix the first point first and see if that issue still holds

**Comment by rudolfbyker** [comment]
Seems to work, both in the IDE and with Mypy on the CLI.

One bit of weirdness, which I can totally live with (maybe it is even desired, although I am not sure), is that when I CTRL+CLICK on a `dicti(...)` call, it takes me directly to the `dict(...)` builtin, as if `dicti=dict`.

File: test/typing_checks.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+v: V = V()
+d: Dict[str, V] = {}
+o: OrderedDict[str, V] = OrderedDict()
+
+
+# ----------------------------------------
+# dicti
+# ----------------------------------------
+
+_: Dict[str, V] = dicti()
```
Comment (rudolfbyker) [inline test/typing_checks.py:19]: This is such a nice way to write type tests! I would not have thought of it...

**Review by rudolfbyker — APPROVED** [review]

File: test/typing_checks.py (hunk 1 above)
Comment (coldfix) [reply test/typing_checks.py:19]: Thanks! Not sure it's the recommended way or if it has any downsides though...

**Comment by coldfix** [comment]
> maybe it is even desired, although I am not sure

Not really, I guess, but good enough for now.

**Pull request merged**